crates/nu-protocol/src/value/mod.rs

```diff
@@ -617,18 +617,23 @@ impl Value {
                     }
                     Value::List { vals, span } => {
                         let mut output = vec![];
+                        let mut hasvalue = false;
                         for val in vals {
-                            output.push(val.clone().follow_cell_path(&[PathMember::String {
+                            let temp = val.clone().follow_cell_path(&[PathMember::String {
                                 val: column_name.clone(),
                                 span: *origin_span,
-                            }])?);
-                            // if let Value::Record { cols, vals, .. } = val {
-                            //     for col in cols.iter().enumerate() {
-                            //         if col.1 == column_name {
-                            //             output.push(vals[col.0].clone());
-                            //         }
-                            //     }
-                            // }
+                            }]);
+                            if let Ok(result) = temp {
+                                hasvalue = true;
+                                output.push(result);
+                            } else if hasvalue {
+                                output.push(Value::String {
+                                    val: "none".to_string(),
+                                    span: Span { start: 0, end: 0 },
```

Comment:
this isn't going to quite work as it will skip the rows at the beginning that are missing the column.

I think what we tried in Nushell was to grab all the values out, and then check to make sure they weren't all empty. You could use a similar logic here and always send a `Value::Nothing` (don't do `Value::String` here because "none" is valid text from the user).

Then, at the end of the loop, we can check to see if we ever saw a real value. If we didn't, we error.

Reply:
Now I check it at last, can you review it ? thanks

Reply:
@chen244 I tried your PR and it works fine on simple cases I tried.

You have one failing test but that test seems to be testing the bug you fixed. So instead of expecting it to fail you can set it to expect the correct result (column of 20 and $nothing).

Reply:
> @chen244 I tried your PR and it works fine on simple cases I tried.
> 
> You have one failing test but that test seems to be testing the bug you fixed. So instead of expecting it to fail you can set it to expect the correct result (column of 20 and $nothing).

Hello, I have used ShellError::NotFound now. But I have a question. The test is used to test missing_column_error. But My fix does not think it has missed but is a nothing. In the test, it does a "size" command which is like get.Should the test be changed to another one?

Reply:
> @chen244 I tried your PR and it works fine on simple cases I tried.
> 
> You have one failing test but that test seems to be testing the bug you fixed. So instead of expecting it to fail you can set it to expect the correct result (column of 20 and $nothing).

Now I get it an empty column length , but I do not know if it is right

Reply:
> > @chen244 I tried your PR and it works fine on simple cases I tried.
> > You have one failing test but that test seems to be testing the bug you fixed. So instead of expecting it to fail you can set it to expect the correct result (column of 20 and $nothing).
> 
> Now I give it an empty column length , but I do not know if it is right

Reply:
@chen244 You can change the test completely, the current one is not very helpful. You can remove it and add a new test that verifies you can select a column with a missing value. For example, `[ { name: ABC, size: 20 }, { name: HIJ } ].size.1 == $nothing`, use `run_test` instead of `fail_test` and check the return value for `true`.

Reply:
> @chen244 You can change the test completely, the current one is not very helpful. You can remove it and add a new test that verifies you can select a column with a missing value. For example, `[ { name: ABC, size: 20 }, { name: HIJ } ].size.1 == $nothing`, use `run_test` instead of `fail_test` and check the return value for `true`.

I update the test now